### PR TITLE
[AppService]: fix ##8743- az webapp deploy

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
**Description**

#8743- az webapp deploy
**Testing Guide**

az webapp deploy --resource-group ResouceGroup --name AppName --src-path SourcePath --type static --target-path staticfiles/test.txt

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by jvano** [comment]
:shipit:

**Review by shrishrirang — APPROVED** [review]

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -2429,3 +2429,13 @@
     - name: Updates a user entry with the listed roles.
       text: az staticwebapp users update -n MyStaticAppName --user-details JohnDoe --role Contributor
 """
+
+helps['webapp deploy'] = """
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:2433]: Is this not needed for function app?

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 2]
```diff
@@ -129,6 +129,7 @@ def load_command_table(self, _):
         g.custom_show_command('identity show', 'show_identity', validator=validate_app_or_slot_exists_in_rg)
         g.custom_command('identity remove', 'remove_identity', validator=validate_app_or_slot_exists_in_rg)
         g.custom_command('create-remote-connection', 'create_tunnel', exception_handler=ex_handler_factory())
+        g.custom_command('deploy', 'perform_onedeploy', validator=validate_onedeploy_params)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:132]: This command should have 'preview' tag

**Review by panchagnula — CHANGES_REQUESTED** [review]
Changes look good.
but we are missing
1. test cases
2. function app equivalent support - assuming this works for function apps as well.
3. Also add "PREVIEW" tag attribute to this command.

**Comment by Kotasudhakarreddy** [comment]
> Changes look good.
> but we are missing
> 
> 1. test cases
> 2. function app equivalent support - assuming this works for function apps as well.
> 3. Also add "PREVIEW" tag attribute to this command.

working on Testcases. Will address the remaining review comments in my next PR.

**Comment by panchagnula** [comment]
> > Changes look good.
> > but we are missing
> > 
> > 1. test cases
> > 2. function app equivalent support - assuming this works for function apps as well.
> > 3. Also add "PREVIEW" tag attribute to this command.
> 
> working on Testcases. Will address the remaining review comments in my next PR.



> > Changes look good.
> > but we are missing
> > 
> > 1. test cases
> > 2. function app equivalent support - assuming this works for function apps as well.
> > 3. Also add "PREVIEW" tag attribute to this command.
> 
> working on Testcases. Will address the remaining review comments in my next PR.

One more thing to add is --slot support on the commands - especially if API supports this.

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py (hunk 1 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:2433]: Added for Functionapp too.

**Comment by Kotasudhakarreddy** [comment]
@panchagnula - please review

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py (hunk 2 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py]: Added preview tag.

**Review by panchagnula — APPROVED** [review]
LGTM

**Review by qwordy — APPROVED** [review]

**Pull request merged**